### DISCUSSION TITLE
BIRT temporary report document cause a full disk

### DISCUSSION THREAD

**Discussion by pyban** [eclipse-birt/birt#1907] [Q&A]
Each time a BIRT report is generated, it will create a temporary report design file inside the document folder of BIRT.

All these temporary files will only be deleted after 30 mins. Since we have many reports are being generated at the same time, this has caused a full disk space issue.

Is there a way to delete all these temporary files sooner or settings that can prevent BIRT from creating all these temporary files ?

![Screenshot 2024-09-13 122916](https://github.com/user-attachments/assets/a5f809d5-5cfb-4c1a-b0a0-a0e622250a65)

**Comment by speckyspooky** [comment]
Which BIRT version do you use and which kind of output will be created DOC, DOCX, XLSX, PDF,....?

**Reply by pyban** [reply]
BIRT version: 4.6
Output created: PDF

In the meantime, we are also migrating from BIRT 4.6 to BIRT 4.14. The same issue persists in version 4.14.

**Comment by speckyspooky** [comment]
I have investigatd on this topic and there are only 2 ways to solve your problem.
The standard mechanism is that all temporary files will be removed when a new BIRT-session will be created.
With a new session all temporary files will be removed.

Therefore you have 2 options:
- Option 1: you can reduce your sesson timeout from 30min to a lower value - see screen viewer.properties
- Option 2: The way how you create the reports must be started in his own session because then you have a automated clean up based on a start of each new session, which means: each report creation = own report creation session.

![grafik](https://github.com/user-attachments/assets/82da401f-aa7d-4ea6-af4f-7050b18f5d4e)
